core/src/main/java/org/elasticsearch/common/util/Preconditions.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.common.util;
+
+import org.elasticsearch.common.Nullable;
+
+/**
+ * Some useful argument checking utilities, inspired by Guava's Preconditions.
+ */
+public class Preconditions {
+
+    /**
+     * Checks if the precondition argument is met; if not, throws an <code>IllegalArgumentException</code>.
+     */
+    public static void checkArgument(final boolean check, @Nullable final String argName) {
+        if (!check) {
+            StringBuilder msg = new StringBuilder("Invalid argument specified");
+            if (argName != null) {
+                msg.append(" for ").append(argName);
+            }
+            throw new IllegalArgumentException(msg.toString());
+        }
+    }
+
+    /**
+     * Checks that the argument is not null; otherwise, throws an <code>IllegalArgumentException</code>
+     */
+    public static void checkNotNull(final Object obj, @Nullable final String argName) {
```

Comment:
This can probably be replaced by requireNonNull() and a Supplier? Just trying to reduce the need for too many helper classes.